packages/web-components/src/utils/slot-child-class-append.ts

```diff
@@ -0,0 +1,106 @@
+        const theSlotEl:HTMLSlotElement = shadow.querySelector(slotSelector);
+        if (slot.wrapperClass) {
+            const childCount = theSlotEl.assignedElements().length; 
+            if (childCount >= 1){
+                const firstEl:Element = theSlotEl.assignedElements()[0];
+                const firstTag: string|null = firstEl?.tagName.toUpperCase();
+                // this makes the assumption that if there is only one slotted element AND it's a div
+                // that the VFS team has wrapped the content in said div
+                if (firstTag !== "DIV" || childCount > 1) { 
+                   wrapSlotElements(theSlotEl, slot.wrapperClass); 
+                }
+                if (firstTag  === "DIV" && childCount == 1) { firstEl.classList.add(slot.wrapperClass); }
+            } 
+        }  
+        if (slot.selectorClasses){
+            slot.selectorClasses.forEach((tuple:TargetElement)=>{
+                addClassToSlottedElements(theSlotEl,tuple[0],tuple[1], tuple[2]||false);
+            });
+        }
+        if (slot.selectorId){
+            slot.selectorId.forEach((tuple:TargetElement)=>{
+                addIdToSlottedElements(theSlotEl,tuple[0],tuple[1], tuple[2]||false )
+            });
+        }
+    });
+    return;
+}
+
+// Applies classes as described in iterateAddAttrs
+export function addClassToSlottedElements(slotEl:HTMLSlotElement, selector:string, theClass:string, shallow:boolean) : null {
+   const els:Element[] = slotEl.assignedElements();
+   if (els && els.length) { 
+       els.forEach((el:HTMLElement)=>{
+            if (el.matches(selector)){ 
+                el.classList.add(theClass);
+            }
+            if (!shallow){
+                const subEls:NodeListOf<HTMLElement>= el.querySelectorAll(selector);
+                subEls.forEach((subEl:HTMLElement)=>subEl.classList.add(theClass))
+            }
+       }); 
+   }
+   return;
+}
+
+// Applies IDs as described in iterateAddAttrs
+export function addIdToSlottedElements(slotEl:HTMLSlotElement, selector:string, theId:string, shallow:boolean) : null {
+   const els:Element[]= slotEl.assignedElements();
+   if (els && els.length) {
+       for ( let i=0; i<els.length; i++){
+           const el:Element= els[i];
+           if (el.matches(selector)){ 
+               el.id=theId;
+               break; 
+           }
+           if (!shallow){
+                const subEl:HTMLElement = el.querySelector(selector) ;
+                if (subEl){
+                    subEl.id=theId;
+                    break;
+                }
+            }
+       }
+   }
+   return;
+}
+
+// creates a new div element, adds the class 
+// and coreesponding slot.name  to it, 
+// appends it to the list returned by slot.assignedElements()
+// removes the slot attribute from all siblings 
+// and moves them into  the newly created div
+export function wrapSlotElements(slot:HTMLSlotElement, clss:string ) : null{
```

Comment:
```suggestion
export function wrapSlotElements(slot:HTMLSlotElement, class:string) : null {
```

Reply:
it's my understanding that 'class' is a keyword in javascript. would 'aClass'  be more readable?

Reply:
lol you're right, woops. Maybe `classname` would be better here?